readme.md

```diff
@@ -0,0 +1,391 @@
+# @kaliber/routing
+
+This is a routing library for React.
+
+It has been designed with portability (of components), server side rendering and reverse routing in mind.
+
+---
+---
+## Usage
+
+_Please see the `examples` directory for working examples._
+
+Create a route map:
+
+```js
+import { asRouteMap } from '@kaliber/routing'
+
+export const routeMap = asRouteMap({
+  home: '',
+  articles: {
+    path: 'articles',
+
+    list: {
+      path: '',
+      data: fetchArticles,
+    }
+    article: {
+      path: ':article',
+      data: fetchArticle,
+    },
+  },
+  notFound: '*'
+})
+```
+
+Pick a route at the server:
+
+```js
+import { pick } from '@kaliber/routing'
+import { routeMap } from './routeMap'
+
+async function resolve(pathname) {
+  const result = pick(pathname,
+    [routeMap, (params, route) => ({ status: 200, params, route })],
+    [routeMap.notFound, { status: 404 }],
+  )
+  return 'route' in result
+    ? { status: result.status, data: await route.data(params) }
```

Comment:
Dit staat een beetje haaks op wat je zult vinden in de advanced example. Daar is een verschil tussen de data op de route en de async api data gemaakt.

Reply:
Tja, blijft tricky. Dit is een totaal onvolledig voorbeeld om een beetje een idee te geven wat er mogelijk is. Data fetching is geen onderdeel van de library en is enorm afhankelijk van het front-end framework dat gebruikt wordt.

De variant van `advanced` is er 1 die zowel server als client side ondersteunt, die complixiteit wil je niet als eerste in de readme zien.